### ISSUE TITLE
Using ansi_up with noImplicitAny typechecking

### ISSUE THREAD

**Issue by robodl** [drudru/ansi_up#89]
Hello!

I'm having issues using `"ansi_up":  "6.0.2"` while using `noImplicitAny: true` in my tsconfig.json.

```
Error: ../../node_modules/ansi_up/ansi_up.ts:749:15 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.

749         if (! this._url_allowlist[parts[0]])
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: ../../node_modules/ansi_up/ansi_up.ts:762:14 - error TS7006: Parameter 'tmplObj' implicitly has an 'any' type.

762 function rgx(tmplObj, ...subst) {
                 ~~~~~~~


Error: ../../node_modules/ansi_up/ansi_up.ts:762:23 - error TS7019: Rest parameter 'subst' implicitly has an 'any[]' type.

762 function rgx(tmplObj, ...subst) {
                          ~~~~~~~~


Error: ../../node_modules/ansi_up/ansi_up.ts:774:15 - error TS7006: Parameter 'tmplObj' implicitly has an 'any' type.

774 function rgxG(tmplObj, ...subst) {
                  ~~~~~~~


Error: ../../node_modules/ansi_up/ansi_up.ts:774:24 - error TS7019: Rest parameter 'subst' implicitly has an 'any[]' type.

774 function rgxG(tmplObj, ...subst) {
                           ~~~~~~~~
```

Did anyone experience this and have a solution?

**Comment by robodl** [comment]
Update:

I tried to override typings using a custom d.ts file:

```
declare module "ansi_up" {
  class AnsiUp {
    private _url_allowlist: object;
  }

  function rgx(tmplObj: any, ...subst: string[]);
  function rgxG(tmplObj: any, ...subst: string[]);

}
```
but I couldn't get it to work.

I think I figured out what was going on.

I was importing ansi_up like this:
`import { AnsiUp } from "ansi_up/ansi_up";`

but this didn't really fly with the custom typings. After updating the import into `import { AnsiUp } from "ansi_up"`, the custom typings worked.
